Mathlib/Topology/Algebra/Valued/ValuativeRel.lean

```diff
@@ -9,59 +9,100 @@ import Mathlib.Topology.Algebra.WithZeroTopology
 
 /-!
 
-# Valuative Relations as Valued
+# Valuative Relations with a Valuative Topology are Valued
 
-In this temporary file, we provide a helper instance
-for `Valued R Γ` derived from a `ValuativeRel R`,
-so that downstream files can refer to `ValuativeRel R`,
-to facilitate a refactor.
+The `IsValuativeTopology` is described in various equivalent forms as filter bases,
+with helper lemmas to construct it under the `ContinuousConstVAdd R R` assumption.
+When present on a `ValuativeRel R`, this implies `Valued R (ValueGroupWithZero R)`,
+which is provided as an instance.
 
 -/
 
+open ValuativeRel TopologicalSpace Filter Topology Set
+
 namespace IsValuativeTopology
 
-section
+variable {R : Type*} [CommRing R] [ValuativeRel R]
 
-/-! # Alternate constructors -/
+local notation "v" => valuation R
 
-variable {R : Type*} [CommRing R] [ValuativeRel R] [TopologicalSpace R]
+theorem hasBasis_nhds_iff [TopologicalSpace R] :
+    IsValuativeTopology R ↔ ∀ x : R,
+    (𝓝 x).HasBasis (fun _ => True)
+      fun γ : (ValueGroupWithZero R)ˣ => { z | v (z - x) < γ } := by
+  simp_rw [hasBasis_iff, ← neg_add_eq_sub]
+  constructor <;> intro h
+  · simp [mem_nhds_iff]
+  · constructor
+    simp [h]
 
-open ValuativeRel TopologicalSpace Filter Topology Set
+/-- A version mentioning subtraction. -/
+lemma mem_nhds_iff' [TopologicalSpace R] [IsValuativeTopology R] {s : Set R} {x : R} :
```

Comment:
I also think we don't need this if we have `hasBasis_nhds_iff`.

Reply:
And also a left-over `True`

Reply:
But this is over a general nhds, while `hasBasis_nhds_zero` is over `nhds 0`.